src/file/file.ts

```diff
@@ -157,18 +143,66 @@ export class File {
     assertPodName(podName)
 
     const connection = this.accountData.connection
-    const extendedInfo = await getExtendedPodsList(
-      connection.bee,
-      podName,
-      prepareEthAddress(this.accountData.wallet!.address),
-      this.accountData.seed!,
-      connection.options?.downloadOptions,
-    )
+    const extendedInfo = await getExtendedPodsListByAccountData(this.accountData, podName)
 
     const meta = (await getRawMetadata(connection.bee, fullPath, extendedInfo.podAddress)).metadata
     assertRawFileMetadata(meta)
     const data = stringToBytes(JSON.stringify(createFileShareInfo(meta, extendedInfo.podAddress)))
 
     return (await uploadBytes(connection, data)).reference
   }
+
+  /**
+   * Gets shared file information
+   *
+   * @param reference swarm reference with shared file information
+   *
+   * @returns shared file information
+   */
+  async getSharedInfo(reference: string | EncryptedReference): Promise<FileShareInfo> {
+    assertAccount(this.accountData)
+    assertEncryptedReference(reference)
+
+    const sharedInfo = await getSharedInfo(this.accountData.connection.bee, reference)
+    assertFileShareInfo(sharedInfo)
+
+    return sharedInfo
+  }
+
+  /**
+   * Saves shared file to a personal account
+   *
+   * @param podName pod where file is stored
+   * @param parentPath parent path of the file
+   * @param reference swarm reference with shared file information
+   * @param options save options
+   *
+   * @returns saved file metadata
+   */
+  async saveShared(
+    podName: string,
+    parentPath: string,
+    reference: string | EncryptedReference,
+    options?: FileReceiveOptions,
+  ): Promise<FileMetadata> {
+    assertPodName(podName)
+    const sharedInfo = await this.getSharedInfo(reference)
+    const connection = this.accountData.connection
+    const extendedInfo = await getExtendedPodsListByAccountData(this.accountData, podName)
+    const now = getUnixTimestamp()
+    const meta = rawFileMetadataToFileMetadata(sharedInfo.meta)
+    const fileName = options?.name ?? sharedInfo.meta.file_name
+    meta.fileName = fileName
+    meta.accessTime = now
+    meta.modificationTime = now
+    meta.creationTime = now
+    meta.filePath = parentPath
+    meta.podName = podName
+    meta.podAddress = extendedInfo.podAddress
```

Comment:
maybe this could be move into a helper function that overrides sharedInfo meta with the new pod data